DependencyInjection/DoctrineExtension.php

```diff
@@ -143,9 +144,18 @@ protected function dbalLoad(array $config, ContainerBuilder $container)
         $container->setParameter('doctrine.connections', $connections);
         $container->setParameter('doctrine.default_connection', $this->defaultConnection);
 
+        /** @psalm-suppress UndefinedClass */
+        if (interface_exists(Middleware::class)) {
+            $container
+                ->getDefinition('doctrine.dbal.logger')
+                ->replaceArgument(0, null);
+        }
+
```

Comment:
Why was this moved out of `loadMiddlewares`?

Reply:
I moved nothing out of `loadMiddlewares`, `loadMiddleware` has just been added in this PR. What am I missing ?

Reply:
useMiddlewaresIfAvailable was renamed to loadMiddlewares. This condition was there.

Reply:
Ok sorry, it's a subjective choice, in my mind, `loadMiddleware` should only register middlewares, not disable what is replaced by a middleware. But I can move it back in `loadMiddleware` if you think it's better.

Reply:
Point is to guarantee logger argument is replaced only when middlewares are registered. After this split, nothing guarantees it anymore and one can call these separately. Perhaps old name was better then, since `useMiddlewares` doesn't imply it only loads them.

Reply:
The tests guarantee that 🙂 But I understand the need from a code point of view, I moved it back into `useMiddlewaresIfAvailable` (I renamed the method too).